pkg/block/blocktest/adapter.go

```diff
@@ -233,11 +243,47 @@ func testAdapterMultipartUpload(t *testing.T, adapter block.Adapter, storageName
 				multiParts[i].PartNumber = partNumber
 				multiParts[i].ETag = partResp.ETag
 			}
+
+			// List parts after upload
+			listResp, err := adapter.ListParts(ctx, obj, resp.UploadID, block.ListPartsOpts{})
+			if blockstoreType != block.BlockstoreTypeS3 {
+				require.ErrorIs(t, err, block.ErrOperationNotSupported)
+			} else {
+				require.NoError(t, err)
+				require.Equal(t, len(parts), len(listResp.Parts))
+				for i, part := range listResp.Parts {
+					require.Equal(t, multiParts[i].PartNumber, part.PartNumber)
+					require.Equal(t, int64(len(parts[i])), part.Size)
+					require.Equal(t, multiParts[i].ETag, part.ETag)
+					require.False(t, listResp.IsTruncated)
+				}
+			}
+
+			// List parts partial
+			maxParts := int32(2)
+			listResp, err = adapter.ListParts(ctx, obj, resp.UploadID, block.ListPartsOpts{MaxParts: &maxParts})
+			if blockstoreType != block.BlockstoreTypeS3 {
+				require.ErrorIs(t, err, block.ErrOperationNotSupported)
+			} else {
+				require.NoError(t, err)
+				require.Equal(t, int(maxParts), len(listResp.Parts))
+				require.True(t, listResp.IsTruncated)
+				require.Equal(t, strconv.Itoa(int(maxParts)), *listResp.NextPartNumberMarker)
+			}
+
 			_, err = adapter.CompleteMultiPartUpload(ctx, obj, resp.UploadID, &block.MultipartUploadCompletion{
 				Part: multiParts,
 			})
 			require.NoError(t, err)
 
+			// List parts after complete should fail
+			_, err = adapter.ListParts(ctx, obj, resp.UploadID, block.ListPartsOpts{})
+			if blockstoreType != block.BlockstoreTypeS3 {
+				require.ErrorIs(t, err, block.ErrOperationNotSupported)
+			} else {
+				require.NotNil(t, err)
```

Comment:
Prefer to test a specific error here.

Reply:
Unfortunately, these are the block contract unit tests. These tests are generic but the errors are derived from the underlying blockstore and are not aligned.
The upside is we have esti tests to check the specific error codes